plenum/server/monitor.py

```diff
@@ -26,6 +26,51 @@
 logger = getlogger()
 
 
+class RequestMeasurement:
+    """
+    Measure request params (throughput, latency)
+    """
+
+    def __init__(self, inner_window=15, min_cnt=16):
+        self.reqs_in_window = 0
+        self.throughput = 0
+        self.inner_window = inner_window
+        self.min_cnt = min_cnt
+        self.first_ts = time.perf_counter()
+        self.window_start_ts = self.first_ts
+        self.alpha = 2 / (self.min_cnt + 1)
+
+    def add_request(self, ordered_ts):
+        self.moving_window(ordered_ts)
+        if ordered_ts >= self.window_start_ts and ordered_ts <= self.window_start_ts + self.inner_window:
+            self.reqs_in_window += 1
+
+    def moving_average(self, next_val):
+        """
+        Implement exponential moving average
+        """
+        return self.throughput * (1 - self.alpha) + next_val * self.alpha
+
+    def moving_window(self, current_ts):
+        while current_ts >= self.window_start_ts + self.inner_window:
+            self.throughput = self.moving_average(self.reqs_in_window / self.inner_window)
+            self.window_start_ts = self.window_start_ts + self.inner_window
+            self.reqs_in_window = 0
+
+    def get_avg_latency(self):
+        pass
+
+    def get_max_latency(self):
+        pass
+
+    def get_throughput(self, request_time=None):
+        request_time = request_time or time.perf_counter()
```

Comment:
This better be either always injected or separated from this method (in which case it wouldn't mutate it's state)